interfaces/builtin/system_backup.go

```diff
@@ -0,0 +1,62 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const systemBackupSummary = `allows read-only access to the entire system for backups`
+
+const systemBackupBaseDeclarationSlots = `
+  system-backup:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const systemBackupConnectedPlugAppArmor = `
+# Description: Allow read-only access to the entire system
+capability dac_read_search,
+
+# read access to everything except items under /dev, /sys and /proc
+/{,var/lib/snapd/hostfs/}[^dsp]** r,
+/{,var/lib/snapd/hostfs/}{d[^e],s[^y],p[^r]}** r,
```

Comment:
Just a quick thought

Can we instead say
```
deny /var/lib/snapd/hostfs/{dev,sys,proc}/** r,
```
or is this somehow different in semantics?

Reply:
We try to avoid deny rules as much as possible since they can't be undone. That said, we could and this is perhaps ok for hostfs (recall, started as more than just hostfs which is why it is there). The rules as written are accurate though, and I'm not sure of the benefit of adding explicit denies even if the likelihood of running into problems with them is low.